tests/test_project/tests/test_password_history.py

```diff
@@ -100,3 +100,179 @@ def test_change_number_hasher_iterations(self):
                     user_config__iterations=HistoryVeryStrongHasher.iterations).count(),
                 1,
             )
+
+    @override_settings(AUTH_PASSWORD_VALIDATORS=[{
+        'NAME': 'django_password_validators.password_history.password_validation.UniquePasswordsValidator',
+        'OPTIONS': {
+            'lookup_range': 5
+        }
+    }])
+    def test_lookup_range_true_positive(self):
+        # We're considering the last 5 passwords and setting a valid password
+        self.create_user(1)
+        self.user_change_password(user_number=1, password_number=1)
+        self.user_change_password(user_number=1, password_number=2)
+        self.user_change_password(user_number=1, password_number=3)
+        self.assert_password_validation_False(user_number=1, password_number=1)
+        self.assert_password_validation_False(user_number=1, password_number=2)
+        self.assert_password_validation_False(user_number=1, password_number=3)
+        self.assert_password_validation_False(user_number=1, password_number=4)
+
+    @override_settings(AUTH_PASSWORD_VALIDATORS=[{
+        'NAME': 'django_password_validators.password_history.password_validation.UniquePasswordsValidator',
+        'OPTIONS': {
+            'lookup_range': 2
+        }
+    }])
+    def test_lookup_range_true_negative(self):
+        # We're considering the last 2 passwords, setting an invalid password
+        self.create_user(1)
+        self.user_change_password(user_number=1, password_number=1)
+        self.user_change_password(user_number=1, password_number=2)
+        self.assert_password_validation_False(user_number=1, password_number=1)
+        self.assert_password_validation_False(user_number=1, password_number=2)
+        self.assert_password_validation_True(user_number=1, password_number=3)
+        self.user_change_password(user_number=1, password_number=3)
+        self.assert_password_validation_False(user_number=1, password_number=3)
```

Comment:
@spokerman12, can you also check that the validation will fail for password 2 thanks to the 2 entries that will be in the history and update the check in the next line to use password 1 which will also confirm that password 1 was deleted?

Reply:
Yup!